src/Common/src/System/Dynamic/Utils/CacheDict.cs

```diff
@@ -11,42 +12,71 @@ namespace System.Dynamic.Utils
     /// <summary>
     /// Provides a dictionary-like object used for caches which holds onto a maximum
     /// number of elements specified at construction time.
-    /// 
-    /// This class is not thread safe.
     /// </summary>
     internal class CacheDict<TKey, TValue>
     {
-        private readonly Dictionary<TKey, KeyInfo> _dict = new Dictionary<TKey, KeyInfo>();
-        private readonly LinkedList<TKey> _list = new LinkedList<TKey>();
-        private readonly int _maxSize;
+
+        // cache size is always ^2. 
+        // items are placed at [hash ^ mask]
+        // new item will displace previous one at the same location.
+        private readonly int mask;
+        private readonly Entry[] entries;
```

Comment:
I just copied it as-is. There could be many style mismatches here.
I think this falls under the rule of preexisting code, since this is from a bug fix on the desktop runtime. I expect that that at some point the style will be addressed via some style-fixing tool.